doc/README.md

```diff
@@ -18,7 +18,7 @@ To compile the GPflow documentation locally:
 
 2. Install doc dependencies
    ```
-   pip install sphinx sphinx_rtd_theme numpydoc nbsphinx sphinx_autodoc_typehints ipython jupytext jupyter_client ipywidgets
+   pip install sphinx sphinx_rtd_theme numpydoc nbsphinx sphinx-autodoc-typehints sphinx-autopackagesummary ipython jupytext jupyter_client ipywidgets matplotlib sklearn tensorflow_datasets
```

Comment:
do we also need to install all these doc dependencies (`matplotlib`, `tensorflow_datasets`)? isn't there a docs_require.txt or smth we can put here instead? What does RTD use?

Reply:
Building the doc locally (even now, irrespective of this PR) fails unless `matplotlib`, `sklearn` and `tensorflow_datasets` are installed - I guess these are required by the Notebooks in the `doc/source/notebooks` directory. So I added them here as it was pretty frustrating.